src/schema/users.ts

```diff
@@ -1636,6 +1711,79 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return { _: true };
     },
+    addUserCompany: async (
+      _,
+      { email }: { email: string },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      if (!email?.length) {
+        throw new ValidationError('Email is required');
+      }
+      const company = await ctx.con.getRepository(Company).findOneBy({
+        domains: ArrayContains([email.split('@')[1]]),
+      });
+
+      const code = await generateVerifyCode();
+
+      const existingUserCompanyEmail = await ctx.con
+        .getRepository(UserCompany)
+        .findOneBy({
+          email,
+        });
+
+      if (existingUserCompanyEmail) {
+        if (existingUserCompanyEmail.userId !== ctx.userId) {
+          throw new ValidationError('Email already exists');
+        }
+
+        const updatedRecord = { ...existingUserCompanyEmail, code };
+        await ctx.con.getRepository(UserCompany).save(updatedRecord);
```

Comment:
IIRC if we use upsert, it'll update if exists, or insert if not?

Reply:
Hmm not sure on typeorm upsert, I vaguely recall it having some issues.
CC: @capJavert you recall anything for upsert?

(We still need the split query anyway for fetching as we don't want to update someone else's entry)

Reply:
save should also do the same but upsert allows providing on conflict part